sidechain/consensus/aura/src/lib.rs

```diff
@@ -235,35 +241,38 @@ fn proposing_remaining_duration<ParentchainBlock: ParentchainBlockTrait>(
 	std::cmp::min(slot_remaining, proposing_duration)
 }
 
-fn authorities<ValidateerFetcher, P, ParentchainHeader>(
+fn authorities<ValidateerFetcher, P, SignedSidechainBlock, ParentchainHeader>(
 	ocall_api: &ValidateerFetcher,
 	header: &ParentchainHeader,
+	shard: ShardIdentifierFor<SignedSidechainBlock>,
 ) -> Result<Vec<AuthorityId<P>>, ConsensusError>
 where
 	ValidateerFetcher: ValidateerFetch + EnclaveOnChainOCallApi,
 	P: Pair,
+	P::Public: UncheckedFrom<[u8; 32]>,
```

Comment:
Authorities as well are too generic IMO. We could use the same itp_types::AccountId type strictly
too much trait bound jungle here